block/s3/inventory_test.go

```diff
@@ -182,14 +192,21 @@ func TestIterator(t *testing.T) {
 		if len(objects) != len(test.ExpectedObjects) {
 			t.Fatalf("unexpected number of objects in inventory. expected=%d, got=%d", len(test.ExpectedObjects), len(objects))
 		}
-		if !reflect.DeepEqual(objects, test.ExpectedObjects) {
-			t.Fatalf("objects in inventory differrent than expected. expected=%v, got=%v", test.ExpectedObjects, objects)
+		for i, obj := range objects {
+			if obj.Key != test.ExpectedObjects[i] {
+				t.Fatalf("retuned object in index %d different than expected. expected=%s, got=%s", i, test.ExpectedObjects[i], obj.Key)
```

Comment:
```suggestion
				t.Fatalf("returned object at index %d different than expected. expected=%s, got=%s", i, test.ExpectedObjects[i], obj.Key)
```
or even (more Go-ish?)
```suggestion
				t.Fatalf("at index %d: expected=%s, got=%s", i, test.ExpectedObjects[i], obj.Key)
```